diff --git a/src/Entity/Vote.php b/src/Entity/Vote.php
index 1cf46b1..035653b 100644
--- a/src/Entity/Vote.php
+++ b/src/Entity/Vote.php
@@ -35,6 +35,9 @@ use Drupal\votingapi\VoteInterface;
  *     "id" = "id",
  *     "uuid" = "uuid",
  *     "bundle" = "type",
+ *   },
+ *   links = {
+ *     "canonical" = "/entity/vote/{vote}",
  *   }
  * )
  */
@@ -288,4 +291,4 @@ class Vote extends ContentEntityBase implements VoteInterface {
     }
   }
 
-}
\ No newline at end of file
+}
diff --git a/votingapi.module b/votingapi.module
index ce74b03..94ffb36 100644
--- a/votingapi.module
+++ b/votingapi.module
@@ -13,6 +13,9 @@
  */
 
 use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Access\AccessResult;
 
 /**
  * Implements hook_help().
@@ -69,4 +72,19 @@ function votingapi_cron() {
     }
     Drupal::state()->set('votingapi.last_cron', REQUEST_TIME);
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_access for allowing access over REST
+ */
+
+function votingapi_vote_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  return AccessResult::allowed();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access for allowing access over REST
+ */
+function votingapi_vote_create_access(AccountInterface $account, array $context, $entity_bundle) {
+  return AccessResult::allowed();
+}
diff --git a/votingapi.routing.yml b/votingapi.routing.yml
index 3beae3e..bcc008c 100644
--- a/votingapi.routing.yml
+++ b/votingapi.routing.yml
@@ -35,4 +35,12 @@ entity.vote_type.delete_form:
     _entity_form: 'vote_type.delete'
     _title: 'Delete'
   requirements:
-    _entity_access: 'vote_type.delete'
\ No newline at end of file
+    _entity_access: 'vote_type.delete'
+
+entity.vote.canonical:
+  path: '/entity/vote/{vote}'
+  defaults:
+    _entity_view: 'vote.full'
+    _title: 'Vote'
+  requirements:
+    _access: 'TRUE'
